src/compiler/moduleSpecifiers.ts

```diff
@@ -111,6 +115,15 @@ import {
     UserPreferences,
 } from "./_namespaces/ts";
 
+function safeJsonRead(packageJsonPath: string, readFile: (path: string) => string | undefined) {

```

Comment:
There is already helper: `tryParseJson`

Reply:
See: https://github.com/microsoft/TypeScript/pull/55015#discussion_r1296222257

Reply:
I still dont understand why is this different from `tryParseJson() || {}`

Reply:
`tryParseJson` thats in `services\utilities.ts`